go-controller/pkg/allocator/ip/subnet/allocator.go

```diff
@@ -77,6 +77,32 @@ func NewAllocator() *allocator {
 	}
 }
 
+// NewAllocatorForNetwork returns an initialized allocator according to the
+// provided NetInfo
+func NewAllocatorForNetwork(netInfo util.NetInfo) (Allocator, error) {
+	ipAllocator := NewAllocator()
+
+	// TODO: does it even make sense to ask for an IP allocator *if* the
+	// network does not have IPAM ?
+	if util.DoesNetworkRequireIPAM(netInfo) {
```

Comment:
My point is this one is initialized for the network you give it to (i.e. the subnet is added, and the excluded IPs / ranges are excluded).

I do understand your point about leaking details of the network controller here ... But my train of thought was this was OK since I was not adding the util.NetInfo dependency - it's already here . From what I understand, that's what this function does: hand the user an allocated initialized for a given NetInfo.

My motivation was not strong at all: I just wanted to leave the `net` dependency out of the network controller pkg - I thought that was an implementation detail I could hide away. I can do it as you're saying.

Reply:
Ok, so lets adds a bit more detail on the doc text of the method on what this does, like how it just pre-allocates the excluded subnets. At least that way, there is a clear picture of when you would use `NewAllocator` vs `NewAllocatorForNetwork`. I wouldn't check `DoesNetworkRequireIPAM` here, that's for the controller to check.